src/common/posts/PostList.js

```diff
@@ -0,0 +1,50 @@
+import './Post.css';
+import Card from '../card/Card';
+import usePostList from './usePostList';
+
+// contains the displayed list of posts
+// isProfileList: true if list is the user's personal or saved lists
+// isUserSavedList: true if list is the user's saved list
+const PostList = ({ header, Posts }) => {
+  
+  const { posts, handleBookmark, handleShow } = usePostList(header, Posts);
+
+  const isProfileList = header === 'My Posts' || header.split(' ')[0] === 'Saved';
+  const isUserSavedList = header.split(' ')[0] === 'Saved';
+  
+    return (
+      <section className='listsSection'>
+        <h2>{header}</h2>
+        <div className='list'>
+          {posts.map((post, index) => {
+
+            const day = post.start.toLocaleString('default', { weekday: 'short' });
+            const month = ((post.start.getMonth()) + 1).toString().padStart(2, '0');
+            const date = post.start.getDate().toString().padStart(2, '0');
+            const startTime = post.start.toLocaleString('default', { hour: '2-digit', minute: '2-digit' });
+            const endTime = post.end.toLocaleString('default', { hour: '2-digit', minute: '2-digit' });
```

Comment:
It's a bit of an anti-pattern to do this computation from inside a map that's inside a return statement. It just looks a bit worse since then you're left with nested return statements and a block of js computation in the middle of your html/css-style code.

I would recommend hiding this in a separate component. Make a component called `PostCard` or something which takes a post & index as props and use 
`{posts.map((post,index) => <PostCard post={post} index={index} /> )}` here,
and then have `PostCard` do all this day/month/time computation and return a `Card` component.